app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt

```diff
@@ -0,0 +1,118 @@
+package com.geeksville.mesh.ui.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Row
+import androidx.compose.material.Checkbox
+import androidx.compose.material.CheckboxDefaults
+import androidx.compose.material.Divider
+import androidx.compose.material.DropdownMenu
+import androidx.compose.material.DropdownMenuItem
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.geeksville.mesh.R
+import com.geeksville.mesh.database.entity.NodeEntity
+
+@Suppress("LongMethod")
+@Composable
+fun NodeMenu(
+    node: NodeEntity,
+    ignoreIncomingList: List<Int>,
+    isThisNode: Boolean = false,
+    onMenuItemAction: (MenuItemAction) -> Unit,
+    onDismissRequest: () -> Unit,
+    expanded: Boolean = false,
+) {
+    val isIgnored = ignoreIncomingList.contains(node.num)
+
+    DropdownMenu(
+        modifier = Modifier.Companion.background(MaterialTheme.colors.background.copy(alpha = 1f)),
+        expanded = expanded,
+        onDismissRequest = onDismissRequest,
+    ) {
+
+        if (!isThisNode) {
+            DropdownMenuItem(
+                onClick = {
+                    onDismissRequest()
+                    onMenuItemAction(MenuItemAction.DirectMessage)
+                },
+                content = { Text(stringResource(R.string.direct_message)) }
+            )
+            Divider()
+            DropdownMenuItem(
+                onClick = {
+                    onDismissRequest()
+                    onMenuItemAction(MenuItemAction.RequestUserInfo)
+                },
+                content = { Text(stringResource(R.string.request_userinfo)) }
+            )
+            Divider()
+            DropdownMenuItem(
+                onClick = {
+                    onDismissRequest()
+                    onMenuItemAction(MenuItemAction.RequestPosition)
+                },
+                content = { Text(stringResource(R.string.request_position)) }
+            )
+            Divider()
+            DropdownMenuItem(
+                onClick = {
+                    onDismissRequest()
+                    onMenuItemAction(MenuItemAction.TraceRoute)
+                },
+                content = { Text(stringResource(R.string.traceroute)) }
+            )
+            Divider()
+            DropdownMenuItem(
+                onClick = {
+                    onDismissRequest()
+                    onMenuItemAction(MenuItemAction.Remove)
+                },
+                content = { Text(stringResource(R.string.remove)) },
+                enabled = ignoreIncomingList.size < 3 || isIgnored
```

Comment:
copy-paste typo?

Reply:
i was attempting to replicate the logic contained [here](https://github.com/meshtastic/Meshtastic-Android/pull/1386/files#diff-198b8c397de541c4449058d13639ab9cd0b56c3cedd98c99619983819d8b304aL22)